pkg/composer/serviceparser/serviceparser_test.go

```diff
@@ -61,6 +61,14 @@ func TestServicePortConfigToFlagP(t *testing.T) {
 			},
 			expected: "127.0.0.1::80",
 		},
+		{
+			ServicePortConfig: types.ServicePortConfig{
+				HostIP:    "::1",
+				Target:    80,
+				Published: "8080",
+			},
+			expected: "::1:8080:80",
```

Comment:
Probably:
```suggestion
			expected: "[::1]:8080:80",
```

Reply:
net.ParseIP() not accept [] address

Reply:
SplitHostPort / JoinHostPort would accept square brackets.
We should fix whatever code is responsible for parsing these if they cannot currently.

Reply:
@THLIVSQAZ 

Since the problematic part is in `splitParts` in `portutil.go`, I would suggest looking into rewriting it to something like this:

```go

func splitParts(rawport string) (string, string, string) {
	lastIndex := strings.LastIndex(rawport, ":")
	containerPort := rawport[lastIndex+1:]
	hostAddrPort := rawport[:lastIndex]
	if lastIndex == -1 {
		return "", "", containerPort
	}

	addr, port, err := net.SplitHostPort(hostAddrPort)
	if err != nil {
		return "", hostAddrPort, containerPort
	}

	return addr, port, containerPort
}

```

`net.SplitHostPort` will take care of the square bracket notation and return a raw address instead, that will then parse correctly with `net.ParseIP`.

LMK if you are ok getting your PR in that direction - otherwise, I will send one.

Reply:
ok, commit is updated